torch/csrc/jit/init.cpp

```diff
@@ -61,6 +86,22 @@ void initJITBindings(PyObject *module) {
      return py::reinterpret_steal<py::object>(python::unflatten(vars, desc));
    });
 
+   py::class_<GraphExecutor>(m,"GraphExecutor")
+   .def(py::init([](py::function func, std::vector<tracer::TraceInput> inputs){
+     return createExecutorByTracing(true, func, std::move(inputs));
+   }))
+   .def("__call__",[](py::args args) {
+     auto ge = py::cast<GraphExecutor&>(args[0]);
+     auto inputs = py::cast<std::vector<at::Tensor>>(tuple_tail(args));
+     auto outputs = ge.run(std::move(inputs));
+     // if we don't tell pybind these are variables it chokes on the conversion.
+     // TODO: fix conversions to be sane and make sure this works.
+     return std::vector<autograd::Variable>(outputs.begin(), outputs.end());
+   });
+  // hook so we can temporarily get to the non-optimizing version of GraphExecutor
```

Comment:
I was worried about memory leaks, because that exercises a similar pathway to the leak we found before.  I guess we can change it and see what happens. I'd be more confident if we had a way of writing tests for leaks.